router/core/graph_server.go

```diff
@@ -69,10 +79,169 @@ type (
+		// Create runtime metrics exported to OTEL
+		s.runtimeMetrics = rmetric.NewRuntimeMetrics(
+			s.logger,
+			r.otlpMeterProvider,
+			// We track runtime metrics with base router config version
+			// even when we have multiple feature flags
+			append([]attribute.KeyValue{
+				otel.WgRouterConfigVersion.String(s.baseRouterConfigVersion),
+			}, s.baseOtelAttributes...),
+			s.processStartTime,
+		)
+
+		// Start runtime metrics
+		if err := s.runtimeMetrics.Start(); err != nil {
+			return nil, err
+		}
+	}
+
+	// Prometheus metricStore rely on OTLP metricStore
+	if s.metricConfig.IsEnabled() {
+		m, err := rmetric.NewStore(
+			rmetric.WithPromMeterProvider(s.promMeterProvider),
+			rmetric.WithOtlpMeterProvider(s.otlpMeterProvider),
+			rmetric.WithLogger(s.logger),
+			rmetric.WithProcessStartTime(s.processStartTime),
+			// Don't pass the router config version or feature flags here
+			// We scope the metrics to the feature flags and config version in the handler
+			rmetric.WithAttributes(baseOtelAttributes...),
+		)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create metric handler: %w", err)
+		}
+
+		s.metricStore = m
+	}
+
+	if s.registrationInfo != nil {
+		publicKey, err := jwt.ParseECPublicKeyFromPEM([]byte(s.registrationInfo.GetGraphPublicKey()))
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse router public key: %w", err)
+		}
+		s.publicKey = publicKey
+	}
+
+	recoveryHandler := recoveryhandler.New(
+		recoveryhandler.WithLogger(s.logger),
+		recoveryhandler.WithPrintStack(),
+	)
+
+	httpRouter := chi.NewRouter()
+
+	/**
+	* Middlewares
+	 */
+
+	httpRouter.Use(recoveryHandler)
+	httpRouter.Use(rmiddleware.RequestSize(int64(s.routerTrafficConfig.MaxRequestBodyBytes)))
+	httpRouter.Use(middleware.RequestID)
+	httpRouter.Use(middleware.RealIP)
+	httpRouter.Use(cors.New(*s.corsOptions))
+
+	gm, err := s.buildGraphMux(ctx, "", s.baseRouterConfigVersion, routerConfig.GetEngineConfig(), routerConfig.GetSubgraphs())
+	if err != nil {
+		return nil, fmt.Errorf("failed to build base mux: %w", err)
+	}
+
+	featureFlagConfigMap := routerConfig.FeatureFlagConfigs.GetConfigByFeatureFlagName()
+	if len(featureFlagConfigMap) > 0 {
+		s.logger.Info("Feature flags enabled", zap.Strings("flags", maps.Keys(featureFlagConfigMap)))
+	}
+
+	multiGraphHandler, err := s.buildMultiGraphHandler(ctx, gm.mux, featureFlagConfigMap)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build feature flag handler: %w", err)
+	}
+
+	brCompressor := middleware.NewCompressor(5, CustomCompressibleContentTypes...)
+	brCompressor.SetEncoder("br", func(w io.Writer, level int) io.Writer {
+		return br.NewWriterLevel(w, level)
+	})
+
+	/**
+	* A group where we can selectively apply middlewares to the graphql endpoint
+	 */
+	httpRouter.Group(func(cr chi.Router) {
+
+		// We are applying it conditionally because brotli compressing the 3MB playground is very slow
+		cr.Use(middleware.Compress(5, CustomCompressibleContentTypes...))
```

Comment:
Can we disable brotli by default?

Reply:
I would like to remove support in a separate PR. The current implementation is too slow to be used in production.